src/SyncAPIConnector/responses/BaseResponse.cs

```diff
@@ -1,66 +1,76 @@
 using System;
+using ERR_CODE = xAPI.Errors.ERR_CODE;
+using APIReplyParseException = xAPI.Errors.APIReplyParseException;
+using System.Text.Json.Nodes;
 
 namespace xAPI.Responses
 {
-    using JSONAware = Newtonsoft.Json.Linq.JContainer;
-    using JSONObject = Newtonsoft.Json.Linq.JObject;
-    using ERR_CODE = xAPI.Errors.ERR_CODE;
-    using APIReplyParseException = xAPI.Errors.APIReplyParseException;
-    using xAPI.Records;
-
     public class BaseResponse
     {
         private bool? status;
-        private string errorDescr;
+        private string? errorDescr;
         private ERR_CODE errCode;
-        private JSONAware returnData;
-        private string customTag;
+        private JsonNode? returnData;
+        private string? customTag;
 
         public BaseResponse(string body)
         {
-            JSONObject ob;
+            JsonNode? ob;
             try
             {
-                ob = (JSONObject)JSONObject.Parse(body);
-
+                ob = JsonNode.Parse(body);
+            }
+            catch (ArgumentNullException)
+            {
+                throw new APIReplyParseException("JSON Parse exception: body is null");
             }
             catch (Exception x)
             {
                 throw new APIReplyParseException("JSON Parse exception: " + body + "\n" + x.Message);
             }
 
-            if (ob == null)
+            if (ob is null)
             {
                 throw new APIReplyParseException("JSON Parse exception: " + body);
             }
             else
             {
+#if DEBUG
```

Comment:
Please verify is this is acceptable. It simplifies validating the json response content. But Console.Write should probably not be in the library code

Reply:
Yes, should not be there. Pls remove it.

Reply:
removed